typing/type_layout.ml

```diff
@@ -17,57 +17,24 @@ open Types
 type t = layout
 
 module Const = struct
-  type t =
-    | Any
-    | Value
-    | Immediate64
-    | Immediate
-    | Void
-
-  let rec constrain_sort_default_void = function
-    | Types.Void -> Void
-    | Types.Value -> Value
-    | Types.Var r ->
-      match !r with
-      | Some sort -> constrain_sort_default_void sort
-      | None -> (r := Some Types.Void; Void)
-
-  let constrain_default_void = function
-    | Types.Any -> Any
-    | Types.Sort sort -> constrain_sort_default_void sort
-    | Types.Immediate64 -> Immediate64
-    | Types.Immediate -> Immediate
-
-  let can_make_void l = Void = constrain_default_void l
+  type t = Layout.const
+
+  let constrain_default_void = Layout.get_defaulting ~default:Sort.Void
+  let can_make_void l = Layout.Void = constrain_default_void l
```

Comment:
I think I would just eliminate the `Const` module from this file, at this point, and move the two functions to the top level - it existed mainly to hold the type.

Reply:
OK